alm/src/utils/getFinalizationEvent.ts

```diff
@@ -31,27 +34,21 @@ export const getFinalizationEvent = async (
   if (!contract || !web3 || !waitingBlocksResolved) return
   // Since it filters by the message id, only one event will be fetched
   // so there is no need to limit the range of the block to reduce the network traffic
-  const events: EventData[] = await contract.getPastEvents(eventName, {
-    fromBlock: 0,
-    toBlock: 'latest',
-    filter: {
-      messageId: message.id
-    }
-  })
+  const events: any = await getEvents(contract, eventName, 0, fromHome, undefined, { messageId: message.id })
   if (events.length > 0) {
     const event = events[0]
-    const [txReceipt, block] = await Promise.all([
-      web3.eth.getTransactionReceipt(event.transactionHash),
-      getBlock(web3, event.blockNumber)
-    ])
+    const txReceipt = await web3.eth.getTransactionReceipt(
```

Comment:
can you explain why the change here is needed

Reply:
same reason as for the one with targetBlock; we can't have event.blockNumber if we're using subgraph...